department-of-veterans-affairs/vets-website
# Profile/aw/53547/editing page tweaks testing

## Summary

- Adds fallback content to the edit page for when a field is not found in the query string of the route.
example: `/profile/edit` or `/profile/edit?fieldName=anythingFake` would result in rendering the fallback content
- To test: you can run the mock server with
``` bash
yarn mock-api --responses src/applications/personalization/profile/mocks/server.js
```

- make sure that the feature toggle `profileUseFieldEditingPage` is set to `true`
- run profile locally:
``` bash
yarn watch --env entry=profile
``` 
- view the edit route with various query strings
	- `/profile/edit` or `/profile/edit?fieldName=anythingFake` will load fallback
	- `/profile/edit?fieldName=mobilePhone` should render mobile phone field

- Some xtra things to consider that I may want to revise.
  - This will currently load any of the profile fields other than email and mobile phone, so I may want to ratchet this down to just those fields? They _should_ all work, but our initial rollout will be just for those 2 fields, so it would make sense that the other fields should probably not render, just in case.
  - There is a `returnPath` query string that determines where the user would be returned to when the field update succeeds. If that query string does not exist, or if it is not a route that is found in the available profile routes, then it will default to navigate to the profile root route on success.
    -  Is this a good solution? Should it go somewhere else, or maybe just stay on the page and show a success alert inline if there is no return route provided?
    - An alternative approach that I could explore, would be to match the field to the route that should return to. So if the field being edited is the mobile number, then it should return to Contact Information, but if someone was editing their Preferred Name (assuming that is supported in the future) then the return path for that would be the Personal Information page. I will have to think on this though because that introduces a lot of possibly uneeded complexity.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
https://github.com/department-of-veterans-affairs/va.gov-team/issues/53547

## Testing done

- Updated tests for new fallback content

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | ![Screenshot 2023-05-17 at 8 34 25 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/ec32b1a1-f1d6-4a23-8789-32c59c4023d6) | ![Screenshot 2023-05-17 at 8 34 46 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/28e98d13-d3d0-49c3-bf1e-3bcd8fe64264) |
| Desktop | ![Screenshot 2023-05-17 at 8 34 12 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/52db9059-eb07-4b44-8d00-e7235daff4f0) | ![Screenshot 2023-05-17 at 8 33 46 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/513c163b-d9ef-4796-9aa8-1b248cb18557) |

## What areas of the site does it impact?

Profile - Edit page

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication
